packages/typeorm/src/typeorm.utils.ts

```diff
@@ -89,26 +76,31 @@ export function getEntityManagerToken(
     : `${dataSource.name}EntityManager`
 }
 
-export async function createDataSource(options: TypeOrmOptions): Promise<DataSource> {
-  return lastValueFrom(
-    defer(() => {
-      const dataSource = new DataSource(options as DataSourceOptions)
-      return dataSource.initialize()
-    }).pipe(handleRetry(options.retryAttempts, options.retryDelay, options.toRetry)),
-  )
+export async function createDataSource(options: TypeOrmOptions, shouldInitialize = true): Promise<DataSource> {
+  if (shouldInitialize) {
+    return lastValueFrom(
+      defer(() => {
+        const dataSource = new DataSource(options as DataSourceOptions)
+        return dataSource.initialize()
+      }).pipe(handleRetry(options.retryAttempts, options.retryDelay, options.shouldRetry)),
+    )
+  } else {
+    const dataSource = new DataSource(options as DataSourceOptions)
+    return Promise.resolve(dataSource)
+  }
 }
 
 function handleRetry(
   retryAttempts = 9,
   retryDelay = 3000,
-  toRetry?: (err: unknown) => boolean,
+  shouldRetry?: (err: unknown) => boolean,
 ): <T>(source: Observable<T>) => Observable<T> {
   return <T>(source: Observable<T>) =>
     source.pipe(
       retryWhen((e) =>
         e.pipe(
           scan((errorCount, error: Error) => {
-            if (toRetry && !toRetry(error)) {
+            if (shouldRetry && !shouldRetry(error)) {
```

Comment:
> f `shouldRetry` is `undefined`

If `shouldRetry` is undefined, it will not retry.

Reply:
> The error is expected to be thrown if shouldRetry is undefined(shouldRetry is optional in this function)
If `shouldRetry` is undefined, it will not retry.